bashplotlib/scatterplot.py

```diff
@@ -35,17 +35,43 @@ def _plot_scatter(xs, ys, size, pch, colour, title, cs):
         print(box_text(title, 2 * (len(get_scale(xs, False, size)) + 1)))
 
     print("-" * (2 * (len(get_scale(xs, False, size)) + 2)))
+
+    # Iterate through all squares in the graph. For each square,
```

Comment:
It's worth trying to understand how this double-loop works. It's not necessarily the best way to solve this problem, and there are many alternative approaches one could take. See if you can come up with some.